tools/ci_build/github/apple/objectivec/assemble_pod_package.py

```diff
@@ -0,0 +1,134 @@
+        This directory can be validated (e.g., with `pod lib lint`) and then zipped to create a package for release.
+    """)
+
+    parser.add_argument("--staging-dir", type=pathlib.Path,
+                        default=pathlib.Path("./onnxruntime-mobile-objc-staging"),
+                        help="Staging directory for the Objective-C pod files.")
+    parser.add_argument("--pod-version", required=True,
+                        help="Objective-C pod version.")
+    parser.add_argument("--pod-source-http", required=True,
+                        help="Objective-C pod source http value (e.g., the package download URL).")
+
+    return parser.parse_args()
+
+
+_template_variable_pattern = re.compile(r"@(\w+)@")  # match "@var@"
+
+
+def gen_file_from_template(template_file: pathlib.Path, output_file: pathlib.Path,
+                           variable_substitutions: dict[str, str]):
+    with open(template_file, mode="r") as template:
+        content = template.read()
+
+    def replace_template_variable(match):
+        variable_name = match.group(1)
+        return variable_substitutions.get(variable_name, match.group(0))
+
+    content = _template_variable_pattern.sub(replace_template_variable, content)
+
+    with open(output_file, mode="w") as output:
+        output.write(content)
+
+
+def copy_to_staging_dir(patterns: list[str], staging_dir: pathlib.Path):
+    paths = [path for pattern in patterns for path in _repo_root.glob(pattern)]
+    for path in paths:
+        repo_relative_path = path.relative_to(_repo_root)
+        dst_path = staging_dir / repo_relative_path
+        os.makedirs(dst_path.parent, exist_ok=True)
+        shutil.copy(path, dst_path)
+
+
+def main():
+    args = parse_args()
+
+    staging_dir = args.staging_dir.resolve()
+    print(f"Assembling files in staging directory: {staging_dir}")
+    if staging_dir.exists():
+        print("Warning: staging directory already exists", file=sys.stderr)
+
+    copy_to_staging_dir(
+        [license_file] + source_files + test_source_files + test_resource_files,
+        staging_dir)
+
+    def as_variable_value(patterns: list[str]):
+        return ", ".join([f'"{pattern}"' for pattern in patterns])
+
+    variable_substitutions = {
+        "VERSION": args.pod_version,
+        "SOURCE_HTTP": args.pod_source_http,
+        "LICENSE_FILE": as_variable_value([license_file]),
+        "INCLUDE_DIR_LIST": as_variable_value(include_dirs),
+        "PUBLIC_HEADER_FILE_LIST": as_variable_value(public_header_files),
+        "SOURCE_FILE_LIST": as_variable_value(source_files),
+        "TEST_SOURCE_FILE_LIST": as_variable_value(test_source_files),
+        "TEST_RESOURCE_FILE_LIST": as_variable_value(test_resource_files),
+    }
+
+    podspec_template = _script_dir / "onnxruntime-mobile-objc.podspec.template"
+    podspec = staging_dir / "onnxruntime-mobile-objc.podspec"
+    gen_file_from_template(podspec_template, podspec, variable_substitutions)
+
```

Comment:
Curious, do you want to create the zip archive here as well?

Reply:
I think zipping and uploading would happen at a later point, after testing the package. Once we have this staging directory, we can run `pod lib lint` in it to test it.